### PR TITLE
DI improvements

### PR DESCRIPTION
**Pull request by Zales0123**
I've also tried to fix a build on Symfony 3.4, but failed :( Do you think you can take a look at it? Something weird is going on there 😄 With my proposed changes and build fixed, it would be ready to list on our official plugins list, I think :)

### CODE REVIEW

**Comment by stefandoorn** [comment]
Thanks! I recall from earlier that we had to use the container in the contexts due to some circular reference issue that back then we didn't get solved. I hope that's none of an issue now anymore. I'll probably try to look into the S3.4 build first (as in another new plugin it doesn't give issues..) and then try to merge this one on top of it. WDYT?

File: tests/Application/src/Repository/ProductRepository.php [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 declare(strict_types=1);
 
-namespace StefanDoorn\SyliusSeoUrlPlugin\Repository;
+namespace Tests\Application\StefanDoorn\SyliusSeoUrlPlugin\Repository;
```
Comment (stefandoorn) [inline tests/Application/src/Repository/ProductRepository.php]: I added this one here so people could use it directly if they don't want to extend the repository themselves, see README. WDYT?

**Comment by stefandoorn** [comment]
Explicitly creating the db leads to other problems, a query on the 'schema create' step: https://travis-ci.org/stefandoorn/SyliusSeoUrlPlugin/jobs/469702150

I highly expect this to be related to the contexts doing queries.. But not sure where, why and how.

**Comment by stefandoorn** [comment]
Ok, 3.4 build is fixed on `master` branch, @Zales0123.

File: tests/Application/src/Repository/ProductRepository.php (hunk 1 above)
Comment (Zales0123) [reply tests/Application/src/Repository/ProductRepository.php]: Seems reasonable, but I think usually people use more plugin than one and at the end, have all models, repositories etc. overridden in their `src/` folder anyway... But I don't insist, I think it's a matter of taste, the README explains the approach properly 👌 I'll remove that

**Comment by stefandoorn** [comment]
Thanks @Zales0123!

**Pull request merged**